cmd/neofs-node/config/engine/shard/config.go

```diff
@@ -68,3 +71,25 @@ func (x *Config) RefillMetabase() bool {
 		"resync_metabase",
 	)
 }
+
+// Mode return value of "mode" config parameter.
+//
+// Panics if read value is not one of predefined
+// shard modes.
+func (x *Config) Mode() (m shard.Mode) {
+	s := config.StringSafe(
+		(*config.Config)(x),
+		"mode",
+	)
+
+	switch s {
+	case "read-write", "":
+		m = shard.ModeReadWrite
+	case "read-only":
+		m = shard.ModeReadOnly
```

Comment:
Do we plan to support any more modes? IMO having `readonly: true` is somewhat more common.

Reply:
I think @realloc has many more mode example for the neofs-next: degraded mode, fault, evacuation.

Reply:
In the future we'd like to have at least:
- read-write // Normal
- read-only // Both MB and BS are working, but in R/O.
- read-only-purge // Can only delete data. When MB or BS are temporary full
- fault // Both Mb and BS are not functional
- degraded-read-write // MB is lost, operations limited, but can slowly accept writes
- degraded-read-only // MB is lost, R/O with limited operations
- degraded-read-only-meta // BS lost, only MB in R/O